hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -3538,18 +3538,31 @@ public void updateRegionsInTransitionMetrics() {
     int totalRITs = 0;
     int totalRITsOverThreshold = 0;
     long oldestRITTime = 0;
+    HashMap<String, RegionState> ritsOverThreshold = null;
     int ritThreshold = this.server.getConfiguration().
       getInt(HConstants.METRICS_RIT_STUCK_WARNING_THRESHOLD, 60000);
     for (RegionState state: regionStates.getRegionsInTransition()) {
       totalRITs++;
       long ritTime = currentTime - state.getStamp();
       if (ritTime > ritThreshold) { // more than the threshold
         totalRITsOverThreshold++;
+        if (ritsOverThreshold == null) {
+          ritsOverThreshold = new HashMap<String, RegionState>();
+        }
+        ritsOverThreshold.put(state.getRegion().getEncodedName(), state);
       }
       if (oldestRITTime < ritTime) {
         oldestRITTime = ritTime;
       }
     }
+    if (LOG.isDebugEnabled() && ritsOverThreshold != null && !ritsOverThreshold.isEmpty()) {
+      StringBuilder sb = new StringBuilder();
+      for (String regionName: ritsOverThreshold.keySet()) {
+        sb.append(regionName + ":" + ritsOverThreshold.get(regionName).getState().name() + "\n");
+      }
+      sb.delete(sb.length()-2, sb.length());
```

Comment:
Do we want to remove last appended new line char? If so, start index should be `sb.length()-1` right?